### DISCUSSION TITLE
--save-temps in DPC++ with HIP support

### DISCUSSION THREAD

**Discussion by zjin-lcf** [intel/llvm#4752] [Q&A] [answered]
When I try -save-temps with DPC++ with HIP support, the intermediate files are not generated.  Can you reproduce that ?  Thanks.

**Comment by npmiller** [comment, marked as answer]
Yes, `--save-temps` is acting a bit strange.

The intermediate files are still generated, they just don't appear in the current directory unless it's empty, so you can either try to find them in `/tmp` or make sure your current working directory is empty when calling `dpcpp`.

Another thing to note is that because of the way the compilation works, some of the intermediary files will just contain the path to the actual intermediary file which will still be in `/tmp`, so be aware of that but it's pretty easy to retrieve it from there.

**Reply by zjin-lcf** [reply]
Could developers specify a directory for intermediate files with the command options ?  Thanks

**Reply by npmiller** [reply]
There is no option for that, `-save-temps=` can only be `-save-temps=cwd` or `-save-temps=obj`, it's not to specify the output directory, I'm not completely sure what `-save-temps=obj` is meant to do, from what I could find it is supposed to save the intermediate files in the same directory as the output of the compilation, when I tried it it did that but only saved the `.o` file and not any other one. The documentation is a bit lacking.

**Reply by zjin-lcf** [reply]
Okay. It may help compiler engineers with an option to specify a directory for intermediate files.
